In [14]:
import numpy as np
import pandas as pd
from sklearn.preprocessing import LabelEncoder
titanic_df = pd.read_csv("titanic.csv")
titanic_df.head()
class_labels = LabelEncoder()
titanic_df['Name'] = class_labels.fit_transform(titanic_df['Name'].values)
titanic_df['Ticket'] = class_labels.fit_transform(titanic_df['Ticket'].values)
titanic_df['Cabin'] = class_labels.fit_transform(titanic_df['Cabin'].values)
titanic_df['Name'] = class_labels.fit_transform(titanic_df['Name'].values)
titanic_df['Survived'] = class_labels.fit_transform(titanic_df['Survived'].values)
titanic_df['Sex'] = class_labels.fit_transform(titanic_df['Sex'].values)
titanic_df['Embarked'] = class_labels.fit_transform(titanic_df['Embarked'].values)
titanic_df.head()

PassengerId  Survived  Pclass  Name  Sex   Age  SibSp  Parch  Ticket  \
0            1         0       3   108    1  22.0      1      0     523   
1            2         1       1   190    0  38.0      1      0     596   
2            3         1       3   353    0  26.0      0      0     669   
3            4         1       1   272    0  35.0      1      0      49   
4            5         0       3    15    1  35.0      0      0     472   

      Fare  Cabin  Embarked  
0   7.2500    147         2  
1  71.2833     81         0  
2   7.9250    147         2  
3  53.1000     55         2  
4   8.0500    147         2

In [18]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import train_test_split
X = titanic_df.drop(["Survived","Age"], axis = 1)
y = titanic_df['Survived']


X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state = 42)

from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train) 
X_test = sc.fit_transform(X_test)

knn = KNeighborsClassifier(n_neighbors=15) 
knn.fit(X_train, y_train) 

y_pred = knn.predict(X_test)
print(y_pred)
print(type(y_pred))

[0 0 0 1 1 1 1 0 1 1 0 0 0 0 0 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 1 1 0 0 0 0 0 1 0 0 0 0 0 1 1 0 1 0 1 0 1 1 1 0 1 1 0 0 1 0 0 0 1 1 1 1 1
 0 0 0 1 1 1 0 1 1 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 1 0 0 1 1
 0 1 0 1 0 0 0 1 0 0 1 1 0 0 0 1 1 1 0 1 0 0 0 1 1 1 0 0 1 0 1 0 0 0 1 0 0
 1 0 0 0 0 1 0 0 0 1 1 1 0 0 0 1 0 0 0 1 0 0 0 1 0 1 0 0 0 1 0 0 0 0 0 0 1
 0 0 0 0 1 0 0 0 0 1 1 1 0 1 0 0 0 1 1 0 1 0 0 0 0 1 0 1 0 0 0 1 0 1 0 0 1
 0 0 0 1 0 1 1 0 0 1 0 1 0 0 1 1 1 0 0 1 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0
 1 1 1 1 0 0 0 0 1]
<class 'numpy.ndarray'>


In [19]:
print(knn.score(X_test, y_test))

0.8171641791044776


The model I chose was KNN, which I used to create a model that compares numerical values of the dataframe correlated with values of how many people survived on the Titanic using trained and sample data, gettiing a score of 0.817. That means that the model accurately predicted 81.7% of the total predictions correctly, which is a pretty good score. 

In [22]:
def precision_and_recall(calc_type:str, tp: int, tn: int,
                        fp: int, fn: int) -> float:
       if calc_type == "precision":
        value = tp / (tp + fp)
        # false positive are where we predicted someone would have a 
        # condition and they actually dont
        return value
        elif calc_type == "recall":
        value = tp / (tp + fn) 
        # false negatives are where we predicted someone would not have a 
        # condition and they actually do
        return value
        else:
        error_msg = "something happened here. check your inputs"
        return error_msg


SyntaxError: invalid syntax (4184275353.py, line 8)